### PR TITLE
Feature/delete events

### LINKED ISSUE #206: Allow DELETE of an erroneous event using the form.
In addition to setting an event cancelled/postponed, there is a need for DELETing a created event in the case of 
a) duplicate events
b) events erroneously or mistakenly published.

Here, setting the event as cancelled is definitely not the right way of informing the public.

The UI must be clear on the distinction, and before allowing the user to DELETE their event, it must list all subevents related to the event, and inform the user that in case of a cancelled or postponed actual event, DELETE must not be used.

DELETE should be reserved for those use cases where the event information is clearly erroneous or unnecessary.

### PR DESCRIPTION
**Pull request by jonraem**
Fixes #206.

### CODE REVIEW

**Comment by jonraem** [comment]
Still needs a change to warning text. Does this need localization? (Other modals in Editor are not localized.)

**Comment by jonraem** [comment]
Ready for review.

**Comment by Rikuoja** [comment]
What I did is I added soft deleting to the API. This way, actually deleting all the event should be allowed as is, so no changes are needed on your part. I will still review this PR shortly.

File: src/validation/validationRules.js [hunk 1]
```diff
@@ -256,7 +256,26 @@ var validations = {
     },
     hasPrice: function hasPrice(values, value, key) {
```
Comment (Rikuoja) [inline src/validation/validationRules.js]: For some reason, the validation error for this validator doesn't disappear even though the data in missing languages have been added.

Still, when trying to submit after filling in the data in required languages, submission does succeed, so this is just a display glitch for the validation error.

**Comment by jonraem** [comment]
Price validation fixed.

**Pull request merged**